### PR TITLE
Update libigl to 2.0.0-e33f534-p1

### PR DESCRIPTION
**Pull request by rbsheth**
* I've followed [this guide](https://docs.hunter.sh/en/latest/creating-new/update.html)
  step by step carefully. **Yes**

* I've tested this package remotely and have excluded all broken builds.
  Here is the links to the Travis/AppVeyor with status "All passed":

  * https://ci.appveyor.com/project/rbsheth/hunter/builds/25541762
  * https://travis-ci.com/rbsheth/hunter/builds/116891995

* For reference, *long* builds where some toolchains failed:
  * https://ci.appveyor.com/project/rbsheth/hunter/builds/25537275
  * https://travis-ci.com/rbsheth/hunter/builds/116865980

* This update will break few old toolchains.
  They are excluded in this pull request: https://github.com/ingenue/hunter/pull/374

The old `libigl` package simply copied the headers+implementation files and did not build the actual code. By turning on the `LIBIGL_USE_STATIC_LIBRARY` flag, normal library build behavior is enforced. @ruslo

### CODE REVIEW

File: cmake/projects/libigl/hunter.cmake [hunk 1]
```diff
@@ -15,18 +15,18 @@ hunter_add_version(
     PACKAGE_NAME
     libigl
     VERSION
-    2.0.0-e33f534-p0
+    2.0.0-e33f534-p1
```
Comment (ruslo) [inline cmake/projects/libigl/hunter.cmake]: Please add a new entry, no need to remove old:
- https://docs.hunter.sh/en/latest/creating-new/update.html#update-package

Comment (rbsheth) [reply cmake/projects/libigl/hunter.cmake]: Fixed, although the previous version is not working.

**Comment by ruslo** [comment]
Testing:
- https://ci.appveyor.com/project/ingenue/hunter/builds/25593471
- https://travis-ci.org/ingenue/hunter/builds/551430107?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.23.195

**Pull request merged**